packages/strapi-plugin-sentry/config/functions/bootstrap.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+module.exports = async () => {
+  // Initialize the Sentry service exposed by this plugin
+  const { sentry } = strapi.plugins.sentry.services;
+  sentry.init();
+
+  // Create a middleware to intercept API errors
+  strapi.app.use(async (ctx, next) => {
+    try {
+      await next();
+    } catch (error) {
+      sentry.sendError(error, (scope, sentryInstance) => {
+        scope.addEventProcessor(event => {
+          // Parse Koa context to add error metadata
+          return sentryInstance.Handlers.parseRequest(event, ctx.request);
```

Comment:
What I would suggest is parsing in as much information as we can, not that my example is great, but on a personal project I created a custom sentry middleware.

https://github.com/canonn-science/CAPIv2-Strapi/blob/5f5b78f8e04b387ce5e15906850b1768a1a0eb01/middlewares/sentry/index.js#L18-L27

What I'm pull in is the Strapi environment which distributes the issues in the sentry dashboard:

![image](https://user-images.githubusercontent.com/8593673/102661456-b0c78200-4131-11eb-8a80-62ed3807bc45.png)

And other various metadata:

![image](https://user-images.githubusercontent.com/8593673/102661495-c9d03300-4131-11eb-925c-5359e6bfe8e7.png)

![image](https://user-images.githubusercontent.com/8593673/102661562-ea988880-4131-11eb-911f-5b181689af7f.png)

![image](https://user-images.githubusercontent.com/8593673/102661639-1ddb1780-4132-11eb-8e33-cd537961b00e.png)

Mine is a messy bit of code, but it could be done properly, regardless, the extra metadata really helps to track down issues especially in larger applications.

Reply:
Might be best to have some configuration options to enable/disable certain reporting